.github/workflows/build_and_run_test_app.yml

```diff
@@ -0,0 +1,111 @@
+name: Build and Run Test Application
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    name: Build Test App
+    strategy:
+      matrix:
+        idf_ver: ["release-v4.3", "release-v4.4", "latest"]
+        idf_target: ["esp32", "esp32c3", "esp32s3"] # @todo ESP32S2 has less RAM and the test_app will not fit
+        exclude:
+          - idf_ver: "release-v4.3"
+            idf_target: esp32s3 # ESP32S3 support started with version 4.4
+    runs-on: ubuntu-20.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: 'true'
+      - name: Build Test Application
+        env:
+          IDF_TARGET: ${{ matrix.idf_target }}
+        shell: bash
+        working-directory: test_app
+        run: |
+          . ${IDF_PATH}/export.sh
+          idf.py build
+      - name: Merge binaries
+        working-directory: test_app/build
+        run: |
+          pip install esptool
```

Comment:
Don't we already have esptool.py in the container? Or is the issue that esptool.py in release/v4.3 doesn't support `merge_bin`?

Reply:
In the case of outdated esptool it should be `pip install -U esptool`

Reply:
works fine with calling `esptool.py` directly.

I just didn't realize that I must run `. ${IDF_PATH}/export.sh` for each step explicitly...